### PR TITLE
CONFIGURE: Some OpenBSD tweaks + let the endianness check build if there's no __attribute__

### PR DESCRIPTION
**Pull request by dwatteau**
The commits are unrelated but they're small changes to the same file.

## OpenBSD

Contrary to FreeBSD, OpenBSD completely removed its `/dev/sequencer` support back in 2013, since sndio is its replacement (for which we already have a MIDI backend):
https://marc.info/?l=openbsd-cvs&m=136333866832000&w=2

So the first commit moves OpenBSD to its own section and always disables SEQ MIDI there, as was recently done for macOS in commit 1aabe9087e815efa56c9692c5b856b9e199528d6.

The other OpenBSD commit just disables `-pedantic` there, because some system headers make the build noisy, otherwise. (It builds fine with `-std=c++11`, though.)

(Basically, this is a backport of what the official OpenBSD port has been doing for some years; I'm trying to reduce the number of patches which are applied in the ports tree.)

## `__attribute__`

The other commit is related to the endianness test in `configure`: it has some `__attribute((used))` lines coming [from Fedora](https://src.fedoraproject.org/rpms/scummvm/blob/f33/f/gcc-lto-2.2.0.patch) since they use LTO by default (well, the check itself could maybe be rewritten so that it will never trigger any dead-code elimination, and remove EBCDIC support ;) but `__attribute__` is used unconditionally so it could fail compiling with some non-GCC-compatible compilers.  In practice, `configure` implicitly targets GCC-compatible compilers I believe, but…

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -2336,6 +2339,9 @@ fi
 #
 echo_n "Checking endianness... "
 cat > tmp_endianness_check.cpp << EOF
+#ifndef __attribute__
```
Comment (lephilousophe) [inline configure:2342]: This doesn't work as __attribute__ is not defined as a macro.
You are basically voiding __attribute__((used)) here.

Comment (dwatteau) [reply configure]: Oops, indeed, sorry for that. Now using `__GNUC__` instead.

Comment (lephilousophe) [reply configure]: Another way could be to use C++11 attribute specifiers but until C++17 it is not guaranteed that an unknown attribute would be ignored so this way is fine.

**Comment by sev-** [comment]
Thank you!

**Pull request merged**